### PR TITLE
fs/unionfs: Support fchstat and chstat callback

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Follow up PR: #4153

## Impact
New feature

### CODE REVIEW

File: fs/unionfs/fs_unionfs.c [hunk 1]
```diff
@@ -1285,6 +1325,45 @@ static int unionfs_fstat(FAR const struct file *filep, FAR struct stat *buf)
   return ops->fstat ? ops->fstat(&uf->uf_file, buf) : -EPERM;
 }
 
+/****************************************************************************
+ * Name: unionfs_fchstat
+ *
+ * Description:
+ *   Change information about an open file associated with the file
+ *   descriptor 'fd',.
```
Comment (gustavonihei) [inline fs/unionfs/fs_unionfs.c:1333]:
```suggestion
 *   structure 'filep'.
```

File: fs/unionfs/fs_unionfs.c [hunk 2]
```diff
@@ -2441,6 +2520,50 @@ static int unionfs_stat(FAR struct inode *mountpt, FAR const char *relpath,
...
+ ****************************************************************************/
+
+static int unionfs_chstat(FAR struct inode *mountpt, FAR const char *relpath,
+                          FAR const struct stat *buf, int flags)
+{
+  FAR struct unionfs_inode_s *ui;
+  FAR struct unionfs_mountpt_s *um;
+  int ret;
+
+  finfo("relpath: %s\n", relpath);
```
Comment (gustavonihei) [inline fs/unionfs/fs_unionfs.c:2534]: This debug log should be moved after the `DEBUGASSERT` for `relpath != NULL`.

File: fs/unionfs/fs_unionfs.c [hunk 3]
```diff
@@ -1285,6 +1325,45 @@ static int unionfs_fstat(FAR const struct file *filep, FAR struct stat *buf)
...
+ * Name: unionfs_fchstat
+ *
+ * Description:
+ *   Change information about an open file associated with the file
+ *   descriptor 'fd',.
+ *
+ ****************************************************************************/
+
+static int unionfs_fchstat(FAR const struct file *filep,
+                           const FAR struct stat *buf, int flags)
```
Comment (gustavonihei) [inline fs/unionfs/fs_unionfs.c:1338]:
```suggestion
                           FAR const struct stat *buf, int flags)
```
nit: just to keep consistency on the order of the qualifiers

File: fs/unionfs/fs_unionfs.c [hunk 4]
```diff
@@ -1285,6 +1325,45 @@ static int unionfs_fstat(FAR const struct file *filep, FAR struct stat *buf)
...
+{
+  FAR struct unionfs_inode_s *ui;
+  FAR struct unionfs_file_s *uf;
+  FAR struct unionfs_mountpt_s *um;
+  FAR const struct mountpt_operations *ops;
+
+  finfo("filep=%p buf=%p\n", filep, buf);
```
Comment (gustavonihei) [inline fs/unionfs/fs_unionfs.c:1345]: This log should be moved after the `DEBUGASSERT` for `filep != NULL`.

File: fs/unionfs/fs_unionfs.c (hunk 2 above)
Comment (xiaoxiang781216) [reply fs/unionfs/fs_unionfs.c:2534]: It's safe here since the pointer isn't dereferenced.

File: fs/unionfs/fs_unionfs.c (hunk 1 above)
Comment (xiaoxiang781216) [reply fs/unionfs/fs_unionfs.c]: Done.

File: fs/unionfs/fs_unionfs.c (hunk 3 above)
Comment (xiaoxiang781216) [reply fs/unionfs/fs_unionfs.c]: Done.

File: fs/unionfs/fs_unionfs.c (hunk 2 above)
Comment (gustavonihei) [reply fs/unionfs/fs_unionfs.c:2534]: Ok, I've checked that our internal vsprintf implementation checks for a NULL pointer and prints `(null)`.

**Review by gustavonihei — APPROVED** [review]
LGTM

**Pull request merged**